src/game/SpellEffects.cpp

```diff
@@ -7147,10 +7155,20 @@ void Spell::EffectSummonDeadPet(SpellEffectIndex /*eff_idx*/)
 {
     if (m_caster->GetTypeId() != TYPEID_PLAYER)
         return;
+
+    bool wasDespawned = false;
     Player* _player = (Player*)m_caster;
     Pet* pet = _player->GetPet();
     if (!pet)
-        return;
+    {
+        pet = new Pet;
+        if (!pet->LoadPetFromDB(_player, 0, 0, false, true))
```

Comment:
The delete should never occur, since its all checked in checkCast, this is simply cleaning code in case of some sort of fatal failure. (someone playing in debugger for example, or missing creature_template entry for pet already in DB)